### PR TITLE
Merging ongoing develop to master

### PR DESCRIPTION
**Pull request by Imunhatep**
Merging ongoing develop to master

### CODE REVIEW

File: tests/GitElephant/Command/DiffCommandTest.php [hunk 1]
```diff
@@ -25,7 +24,7 @@
 class DiffCommandTest extends TestCase
 {
     /**
-     * @var \GitElephant\Command\DiffCommand;
+     * @var \GitElephant\Command\DiffCommand ;
```
Comment (GenieTim) [inline tests/GitElephant/Command/DiffCommandTest.php]:
```suggestion
     * @var \GitElephant\Command\DiffCommand
```

File: tests/GitElephant/Command/LsTreeCommandTest.php [hunk 2]
```diff
@@ -26,7 +25,7 @@
 class LsTreeCommandTest extends TestCase
 {
     /**
-     * @var \GitElephant\Command\LsTreeCommand;
+     * @var \GitElephant\Command\LsTreeCommand ;
```
Comment (GenieTim) [inline tests/GitElephant/Command/LsTreeCommandTest.php]:
```suggestion
     * @var \GitElephant\Command\LsTreeCommand
```

File: src/GitElephant/Command/SubCommandCommand.php [hunk 3]
```diff
@@ -107,7 +107,7 @@ public function getCommand()
             $command .= ' ';
         }
         $subjects = $this->getCommandSubjects();
-        if (count($subjects) > 0) {
+        if ((is_countable($subjects) ? count($subjects) : 0) > 0) {
```
[the remark this answers is not in the document]
Comment (Imunhatep) [reply src/GitElephant/Command/SubCommandCommand.php]: Exactly, seen it on initial change, but in absence of returned typehint, its changed to better )
Anyway, this should be addressed on develop branch, this MR only syncs dev with master branches.

File: src/GitElephant/Objects/Commit.php [hunk 4]
```diff
@@ -269,25 +275,25 @@ private function parseOutputLines($outputLines)
      *
      * @return bool
      */
-    public function isRoot()
+    public function isRoot(): bool
     {
-        return count($this->parents) == 0;
+        return emtpy($this->parents);
```
Comment (GenieTim) [inline src/GitElephant/Objects/Commit.php]: Sorry, typo crashing tests
```suggestion
        return empty($this->parents);
```

**Pull request merged**